processing/src/main/java/org/apache/druid/segment/vector/VectorSelectorUtils.java

```diff
@@ -47,14 +47,33 @@
       retVal = new boolean[offset.getMaxVectorSize()];
     }
 
-    // Probably not super efficient to call "get" so much, but, no worse than the non-vectorized version.
     if (offset.isContiguous()) {
+      final int startOffset = offset.getStartOffset();
+      nullIterator.advanceIfNeeded(startOffset);
+      if (!nullIterator.hasNext()) {
+        return null;
+      }
       for (int i = 0; i < offset.getCurrentVectorSize(); i++) {
-        retVal[i] = nullValueBitmap.get(i + offset.getStartOffset());
+        final int row = i + startOffset;
+        nullIterator.advanceIfNeeded(row);
+        if (!nullIterator.hasNext()) {
+          break;
+        }
+        retVal[i] = row == nullIterator.peekNext();
```

Comment:
since the vectorized version doesn't save the `nullMark` like the other version, doing `next` here could consume a value that is outside of the range we are building the vector for, and we would lose it for the next vector that has that value.